src/pages/POS/POS.vue

```diff
@@ -971,6 +997,122 @@ export default defineComponent({
         });
       }
     },
+    async handleBatchSelected(batchName: string) {
+      if (!this.pendingBatchItem) {
+        return;
+      }
+
+      const { item, quantity } = this.pendingBatchItem;
+      this.pendingBatchItem = null;
+
+      try {
+        const isInventoryItem = await this.fyo.getValue(
+          ModelNameEnum.Item,
+          item.name,
+          'trackItem'
+        );
+        if (!this.batchAddedItems.includes(item.name)) {
+          this.batchAddedItems.push(item.name);
+        }
+
+        let availableQty = 0;
+        if (isInventoryItem) {
+          availableQty =
+            (await fyo.db.getStockQuantity(
+              item.name,
+              undefined,
+              undefined,
+              undefined,
+              batchName
+            )) ?? 0;
+
+          const itemIndex = this.items.findIndex((i) => i.name === item.name);
+          if (itemIndex !== -1) {
+            this.items[itemIndex].availableQty = availableQty ?? 0;
+          }
+        }
+
+        const existingItems =
+          this.sinvDoc.items?.filter(
+            (invoiceItem) =>
+              invoiceItem.item === item.name &&
+              invoiceItem.batch === batchName &&
+              !invoiceItem.isFreeItem
+          ) ?? [];
+
+        const completeItem = (await this.fyo.doc.getDoc(
+          ModelNameEnum.Item,
+          item.name
+        )) as Item;
+
+        await validateQty(
+          this.sinvDoc as SalesInvoice,
+          completeItem,
+          existingItems as InvoiceItem[]
+        );
+
+        const itemsHsncode = (await this.fyo.getValue(
+          'Item',
+          item?.name,
+          'hsnCode'
+        )) as number;
+
+        if (existingItems.length) {
+          const currentQty = existingItems[0].quantity ?? 0;
+          const addQty = quantity ?? 1;
+
+          await existingItems[0].set('quantity', currentQty + addQty);
+        } else {
+          await this.sinvDoc.append('items', {
+            rate: item.rate as Money,
+            item: item.name,
+            quantity: quantity ?? 1,
+            hsnCode: itemsHsncode,
+            batch: batchName,
+          });
+        }
```

Comment:
```suggestion
        if (existingItems.length) {
          const currentQty = existingItems[0].quantity ?? 0;
          const addQty = quantity ?? 1;

          await existingItems[0].set('quantity', currentQty + addQty);
        } else {
          await this.sinvDoc.append('items', {
            rate: item.rate as Money,
            item: item.name,
            quantity: quantity ?? 1,
            hsnCode: itemDoc.hsnCode,
            batch: batchName,
          });
        }
```
<br/>
itemDoc is already available, so get hsnCode from it instead of calling getValue